Installation issue: esmf does not set ESMF_COMM when using hpcx-mpi

### Steps to reproduce the issue

Use hpcx-mpi as mpi provider
```console
$ spack add esmf
$ spack install
```
Can be fixed by add hpcx-mpi to https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/esmf/package.py#L193
@climbfuji 

### Error message

Will get a bunch of errors indicate MPI not found
<details><summary>Error message</summary><pre>
>> 2143    /mnt/nfs4/node2/home/jiych/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-11.3.0/parallel-netcdf-1.12.2-j3rv6kyr4efjbfijwfqwmjb6jiugc3vh/include/pnetcdf.h:676:58: error: 'MPI_Info' has not been declared
</pre></details>


### Information on your system

* **Spack:** 0.18.0.dev0 (3f57c7c67ac29a4dbbc7caeb8b9c808b86d1e5fe)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-zen2
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/8550961/spack-build-out.txt)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/esmf/package.py
+++ b/var/spack/repos/builtin/packages/esmf/package.py
@@ -190,7 +190,8 @@ class Esmf(MakefilePackage):
                 # The mpich 3 series split apart the Fortran and C bindings,
                 # so we link the Fortran libraries when building C programs:
                 os.environ['ESMF_CXXLINKLIBS'] = '-lmpifort'
-            elif '^openmpi' in spec:
+            elif '^openmpi' in spec or \
+                 '^hpcx-mpi' in spec:
                 os.environ['ESMF_COMM'] = 'openmpi'
             elif '^intel-parallel-studio+mpi' in spec or \
                  '^intel-mpi' in spec or \

